editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/bean/AttributeTextFieldPropertyUICreator.java

```diff
@@ -0,0 +1,90 @@
+
+import org.eclipse.core.databinding.DataBindingContext;
+import org.eclipse.core.databinding.observable.map.IObservableMap;
+import org.eclipse.jface.databinding.swt.WidgetProperties;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.ui.views.properties.tabbed.TabbedPropertySheetWidgetFactory;
+import org.fusesource.ide.camel.editor.properties.creators.AbstractTextFieldParameterPropertyUICreator;
+import org.fusesource.ide.camel.editor.properties.creators.modifylisteners.text.AbstractTextParameterPropertyModifyListener;
+import org.fusesource.ide.camel.model.service.core.catalog.Parameter;
+import org.fusesource.ide.camel.model.service.core.catalog.eips.Eip;
+import org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement;
+import org.fusesource.ide.camel.model.service.core.model.CamelBean;
+import org.w3c.dom.Node;
+
+/**
+ * @author brianf
+ *
+ */
+public class AttributeTextFieldPropertyUICreator extends AbstractTextFieldParameterPropertyUICreator {
+
+	public AttributeTextFieldPropertyUICreator(DataBindingContext dbc, IObservableMap<String, String> modelMap, Eip eip, AbstractCamelModelElement camelModelElement, Parameter parameter,
+			Composite parent, TabbedPropertySheetWidgetFactory widgetFactory) {
+		super(dbc, modelMap, eip, camelModelElement, parameter, parent, widgetFactory, 
+				new AttributeTextParameterPropertyModifyListenerForAdvanced(camelModelElement, parameter));
+	}
+
+	public AttributeTextFieldPropertyUICreator(DataBindingContext dbc, IObservableMap<String, String> modelMap, Eip eip, AbstractCamelModelElement camelModelElement, Parameter parameter,
+			Composite parent, TabbedPropertySheetWidgetFactory widgetFactory, AbstractTextParameterPropertyModifyListener listener) {
+		super(dbc, modelMap, eip, camelModelElement, parameter, parent, widgetFactory, listener);
+	}
+
+	private String findInitialValue(Object value, String defaultValue) {
+		if (value != null && value instanceof String) {
+			return (String) value;
+		} else if (defaultValue != null) {
+			return defaultValue;
+		}
+		return "";
+	}
+	
+	@Override
+	public String getInitialValue() {
+		final String parameterName = parameter.getName();
+		final Object parameterValue = camelModelElement.getParameter(parameterName);
+		final Parameter param = eip.getParameter(parameterName);
+		String defaultValue = null;
+		if (param != null && param.getDefaultValue() != null) {
+			defaultValue = param.getDefaultValue();
+		}
+		if (camelModelElement instanceof CamelBean && parameterValue == null) {
+			final Object attrValue = getAttributeValue(parameterName);
+			return findInitialValue(attrValue, defaultValue);
+		}
+		return findInitialValue(parameterValue, defaultValue);
+	}
+
+	protected Object getAttributeValue(String attrName) {
+		if (camelModelElement.getXmlNode() != null && camelModelElement.getXmlNode().hasAttributes()) {
```

Comment:
what do you think to extract camelModelElement.getXmlNode() in a variable as it is called 3 times?